src/Microsoft.Extensions.Configuration.AzureAppConfiguration/LogHelper.cs

```diff
@@ -74,5 +74,15 @@ public static string BuildPushNotificationUnregisteredEndpointMessage(string res
         {
             return $"{LoggingConstants.PushNotificationUnregisteredEndpoint} '{resourceUri}'.";
         }
+
+        public static string BuildFailoverToDifferentEndpointMessage(string originalEndpoint, string currentEndpoint)
+        {
+            return $"{LoggingConstants.RefreshFailedToGetSettingsFromEndpoint} '{originalEndpoint.TrimEnd('/')}'. {LoggingConstants.RefreshTryingToReadFromEndpoint} '{currentEndpoint.TrimEnd('/')}'.";
```

Comment:
Is the trim necessary?

Reply:
It was the pattern we followed with the original PR that added logging, since the endpoints always have a '/' at the end.

Reply:
Okay.

Parameters need validation for null.

Reply:
Good catch! We need it throughout this helper class.

Reply:
So just for the endpoint parameters because we call `TrimEnd` right?

Reply:
Yes. Without parameter validation it's a possible null-ref.